datahub-web-react/src/app/entity/shared/entity/EntityActions.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from 'react';
+import { Button, message } from 'antd';
+import { LinkOutlined } from '@ant-design/icons';
+import { SearchSelectModal } from '../components/styled/search/SearchSelectModal';
+import { useEntityRegistry } from '../../../useEntityRegistry';
+import { EntityCapabilityType } from '../../Entity';
+
+export enum EntityActionItem {
+    /**
+     * Batch add a Glossary Term to a set of assets
+     */
+    BATCH_ADD_GLOSSARY_TERM,
+    /**
+     * Batch add a Domain to a set of assets
+     */
+    BATCH_ADD_DOMAIN,
+}
+
+interface Props {
+    actionItems: Set<EntityActionItem>;
+    refetchForEntity?: () => void;
+}
+
+function EntityActions(props: Props) {
+    // eslint ignore react/no-unused-prop-types
+    const entityRegistry = useEntityRegistry();
+    const { actionItems, refetchForEntity } = props;
+    const [isBatchAddGlossaryTermModalVisible, setIsBatchAddGlossaryTermModalVisible] = useState(false);
+    const [isBatchSetDomainModalVisible, setIsBatchSetDomainModalVisible] = useState(false);
+
+    // eslint-disable-next-line
+    const batchAddGlossaryTerms = (entityUrns: Array<string>) => {
+        refetchForEntity?.();
+        setIsBatchAddGlossaryTermModalVisible(false);
+        message.success('Successfully added glossary terms!');
+    };
+
+    // eslint-disable-next-line
+    const batchSetDomains = (entityUrns: Array<string>) => {
+        refetchForEntity?.();
```

Comment:
it would be great to track these events with the datahub analytics lib so admins can know how often bulk edits are happening!

Reply:
Yes agree! Not completely implemented yet - Once we have the APIs I'll include those tracking events